bot/seasons/season.py

```diff
@@ -38,48 +43,79 @@ def get_season(bot, season_name: str = None, date: datetime.date = None):
 
     seasons = get_seasons()
 
-    # If there's a season name, we can just grab the correct class.
+    # Use season override if season name not provided
+    if not season_name and Client.season_override:
+        log.debug(f"Season override found: {Client.season_override}")
+        season_name = Client.season_override
+
+    # If name provided grab the specified class or fallback to evergreen.
     if season_name:
         season_name = season_name.lower()
         if season_name not in seasons:
             season_name = 'evergreen'
-
-        season_lib = importlib.import_module(f'bot.seasons.{season_name}')
-        season_class = getattr(season_lib, season_name.capitalize())
+        season_class = get_season_class(season_name)
         return season_class(bot)
 
     # If not, we have to figure out if the date matches any of the seasons.
     seasons.remove('evergreen')
-
     for season_name in seasons:
-        season_lib = importlib.import_module(f'bot.seasons.{season_name}')
-        season_class = getattr(season_lib, season_name.capitalize())
-        season_object = season_class(bot)
-        if season_object.start <= date <= season_object.end:
-            return season_object
+        season_class = get_season_class(season_name)
+        # check if date matches before returning an instance
+        if season_class.start() <= date <= season_class.end():
+            return season_class(bot)
     else:
-        evergreen_lib = importlib.import_module(f'bot.seasons.evergreen')
-        return evergreen_lib.Evergreen(bot)
+        evergreen_class = get_season_class('evergreen')
+        return evergreen_class(bot)
 
 
 class SeasonBase:
     name = None
+    date_format = "%d/%m-%Y"
+
+    @staticmethod
+    def current_year():
+        return datetime.date.today().year
 
-    def __init__(self):
-        current_year = datetime.date.today().year
-        date_format = "%d/%m-%Y"
-        self.start = datetime.datetime.strptime(f"{self.start_date}-{current_year}", date_format).date()
-        self.end = datetime.datetime.strptime(f"{self.end_date}-{current_year}", date_format).date()
+    @classmethod
+    def start(cls):
+        return datetime.datetime.strptime(f"{cls.start_date}-{cls.current_year()}", cls.date_format).date()
+
+    @classmethod
```

Comment:
While functionally the same, I do think a static method is more appropriate for these.

Reply:
It's not functionally the same. A static method does not return the class as an argument.

Reply:
I mean that in the sense that they can both do the same thing. You can always access `cls` by naming it yourself: `SeasonBase.whatever`.

Reply:
Sorry, I disagree that a static method should be used in place of a class method in cases where the class is actually of use within. Class methods are not required to return an instance of it's own class, that's just it's most common usage. I find the usage of a class method more proper and elegant in style.

Regardless though, in this particular case, a class method is definitely required because the class that is being passed as the `cls` argument is not `SeasonBase`: it is the class that has inherited `SeasonBase`, the specific season classes such as Halloween. If I used `SeasonBase` as the origin, it would only result in the attributes `start_date` and `end_date` not being defined.

Reply:
Agree with scragly here, resolving comment.